translator/source_code/combine_wzdx.py

```diff
@@ -0,0 +1,157 @@
+import json
+import copy
+from shapely.geometry.polygon import Polygon
+from shapely.geometry import Point
+import geopy
+from geopy.distance import geodesic
+import pyproj
+import getopt
+import sys
+
+
+def main():
+
+    icone_file, cotrip_file, outputfile = parse_arguments(
+        sys.argv[1:], default_output_file_name='combined_wzdx_message.geojson')
+
+    with open(outputfile, 'w+') as f:
+        wzdx_icone = json.loads(open(
+            icone_file, 'r').read())
+        wzdx_cotrip = json.loads(open(
+            cotrip_file, 'r').read())
+
+        polygon = generate_polygon(
+            wzdx_cotrip['features'][0]['geometry']['coordinates'], 100)
+        feature = iterate_feature(polygon, wzdx_icone)
+        if feature:
+
+            f.write(json.dumps(combine_wzdx(
+                wzdx_cotrip, wzdx_icone, feature), indent=2))
+            print('duplicate iCone and COtrip messages were found and combined WZDx message was written to combined_wzdx_message.geojson file')
+
+        else:
+            print('no duplicate messages were found')
+
+
+def combine_wzdx(wzdx_cotrip, wzdx_icone, icone_feature):
+    combined_wzdx = copy.deepcopy(wzdx_cotrip)
+    combined_wzdx['features'][0]['properties']['vehicle_impact'] = icone_feature['properties']['vehicle_impact']
+    combined_wzdx['road_event_feed_info']['data_sources'].append(
+        wzdx_icone['road_event_feed_info']['data_sources'][0])
+
+    return combined_wzdx
+
+
+def iterate_feature(polygon, wzdx_message):
+    for feature in wzdx_message['features']:
+        for coord in feature['geometry']['coordinates']:
+            if isPointInPolygon(Point(coord[1], coord[0]), polygon):
```

Comment:
Are we saying here that any overlap of the two work zones means it is the same work zone and should be combined? Is there ever a chance of separate work zones that happen to overlap a part of geometry?

Reply:
Our intent was to make a basic algorithm that would work most of the time. I have had a few discussions with tony where we came up with tons of different algorithms, including heading checks, checking whether the iCone linestring crosses any point of the polygon, matching both to roads using a maps api, and even much simpler options like using a rectangle instead of a polygon for increased performance. 
**I would not at all be opposed to implementing a heading check**, I just don't want to spend too much time perfecting this algorithm when we don't have any examples of duplicated workzones. I say that after we already wrote the polygon generator, sunk cost I guess? who knows haha